packages/bff/src/utilities/eserviceDocumentUtils.ts

```diff
@@ -0,0 +1,261 @@
+import { FileManager, WithLogger } from "pagopa-interop-commons";
+import { ApiError, genericError } from "pagopa-interop-models";
+import { P, match } from "ts-pattern";
+import YAML from "yaml";
+import { z } from "zod";
+import { config } from "../config/config.js";
+import {
+  ErrorCodes,
+  interfaceExtractingInfoError,
+  invalidInterfaceContentTypeDetected,
+  invalidInterfaceFileDetected,
+  openapiVersionNotRecognized,
+} from "../model/domain/errors.js";
+import { CatalogProcessClient } from "../providers/clientProvider.js";
+import { BffAppContext } from "../utilities/context.js";
+
+// eslint-disable-next-line max-params
+export async function verifyAndCreateEServiceDocument(
+  catalogProcessClient: CatalogProcessClient,
+  fileManager: FileManager,
+  eService: catalogApi.EService,
+  doc: bffApi.createEServiceDocument_Body,
+  descriptorId: string,
+  documentId: string,
+  ctx: WithLogger<BffAppContext>
+): Promise<void> {
+  const contentType = doc.doc.type;
+  if (!contentType) {
+    throw invalidInterfaceContentTypeDetected(
+      eService.id,
+      "invalid",
+      eService.technology
+    );
+  }
+
+  const serverUrls = await processFile(doc, eService.technology, eService.id);
+  const filePath = await fileManager.storeBytes(
+    {
+      bucket: config.s3Bucket,
+      path: config.eserviceDocumentsPath,
+      resourceId: documentId,
+      name: doc.doc.name,
+      content: Buffer.from(await doc.doc.arrayBuffer()),
+    },
+    ctx.logger
+  );
+
+  const calculateChecksum = async (stream: Readable): Promise<string> =>
+    new Promise((resolve, reject) => {
+      const hash = crypto.createHash("sha256");
+
+      stream.on("data", (data) => {
+        hash.update(data);
+      });
+
+      stream.on("end", () => {
+        resolve(hash.digest("hex"));
+      });
+
+      stream.on("error", (err) => {
+        reject(err);
+      });
+    });
+
+  const checksum = await calculateChecksum(Readable.from(doc.doc.stream()));
+  try {
+    await catalogProcessClient.createEServiceDocument(
+      {
+        documentId,
+        prettyName: doc.prettyName,
+        fileName: doc.doc.name,
+        filePath,
+        kind: doc.kind,
+        contentType,
+        checksum,
+        serverUrls,
+      },
+      {
+        headers: ctx.headers,
+        params: {
+          eServiceId: eService.id,
+          descriptorId,
+        },
+      }
+    );
+  } catch (error) {
+    await fileManager.delete(config.s3Bucket, filePath, ctx.logger);
+    throw error;
+  }
+}
+
+const getFileType = (
+  name: string
+): "json" | "yaml" | "wsdl" | "xml" | undefined =>
+  match(name)
+    .with(P.string.endsWith("json"), () => "json" as const)
+    .with(
+      P.string.endsWith("yaml"),
+      P.string.endsWith("yml"),
+      () => "yaml" as const
+    )
+    .with(P.string.endsWith("wsdl"), () => "wsdl" as const)
+    .with(P.string.endsWith("xml"), () => "xml" as const)
+    .otherwise(() => undefined);
```

Comment:
I suggest asking whether we want to maintain the same behavior in the [Scala codebase](https://github.com/pagopa/interop-be-backend-for-frontend/blob/b99cc2f837c6ef49a2eebf757bb74ecf4bf5e9fc/src/main/scala/it/pagopa/interop/backendforfrontend/common/system/FileManagerUtils.scala#L32), where logs detect the type if debug mode is activated.
```suggestion
const getFileType = (
  name: string
): "json" | "yaml" | "wsdl" | "xml" | undefined => {

 const detectedType = match(name)
    .with(P.string.endsWith("json"), () => "json" as const)
    .with(
      P.string.endsWith("yaml"),
      P.string.endsWith("yml"),
      () => "yaml" as const
    )
    .with(P.string.endsWith("wsdl"), () => "wsdl" as const)
    .with(P.string.endsWith("xml"), () => "xml" as const)
    .otherwise(() => undefined);
    
    logger.debug(`Detected ${detectedType} interface content type for eservice: ${eService.id}`)
    return detectedType;
```